src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -64,6 +64,10 @@ export class EditorPart extends Part implements EditorGroupsServiceImpl, IEditor
 	private _onDidMoveGroup: Emitter<IEditorGroupView> = this._register(new Emitter<IEditorGroupView>());
 	get onDidMoveGroup(): Event<IEditorGroupView> { return this._onDidMoveGroup.event; }
 
+	private onDidSetGridWidget = this._register(new Emitter<{ width: number; height: number; }>());
```

Comment:
@joaomoreno the purpose of this event seems only to be to trigger `onDidSizeConstraintsChange`. Can we avoid it and just fire that event directly?

Reply:
We can't... the editor part constraints change whenever the grid widget itself changes as well as the grid widget's constraints change. So, we need an event to fire when the grid widget is set and we need a relay event for the grid widget `onDidChange` itself. We must combine them with `anyEvent`.